packages/cosmic-swingset/src/export-kernel-db.js

```diff
@@ -384,13 +384,13 @@ export const spawnSwingStoreExport = (
   cp.on('error', kits.done.reject).on('exit', onExit).on('message', onMessage);
 
   kits.done.promise
-    .catch(err => {
-      kits.started.reject(err);
-    })
     .then(() => {
       cp.off('error', kits.done.reject)
         .off('exit', onExit)
         .off('message', onMessage);
+    })
+    .catch(err => {
+      kits.started.reject(err);
```

Comment:
It's indeed a semantic change, but not as described. The `off` calls need to be made unconditionally of the success or rejection of `kits.done.promise`. By moving the catch to after, the listener would only be unregistered on fulfillment of the promise, not rejection. Since `off` will basically never throw, in think it's safe to just void the chained promise.

Reply:
Thanks, done